app/controllers/bookmarks_controller.rb

```diff
@@ -132,23 +132,22 @@ def status_action documents
     status = params['action']
     Array(documents.map(&:id)).each do |id|
       media_object = MediaObject.find(id)
-      if cannot? :update, media_object
-        errors += ["#{media_object.title} (#{id}) #{t('blacklight.messages.permission_denied')}."]
+      if status == 'publish'
+        success_ids << id if can?(:update, media_object)
```

Comment:
That is best case. But worst case you will get a page with no flash messages. Heres an example of best case:

Lets assume you got 3 media objects (one published, one unpublished and one of another status...like unavailable). The user has permission to unpublish/publish these objects. Therefore you will get two flash messages. One with success and one with alerts. Which is what you expect to happen as you described above.

Now worst case, lets say I got three media objects. Two published and one unpublished. The user has no permissions to any of these objects. So what will happen, first object is grabbed, it will hit the `if status == publish` block, its true, so it will go into the block, it will check if I got permission? I don't. Thefore it exits the if statement and fall all the way until the end of the `each` block. And it will then proceed to grab the next element. Same thing happens. Next element? It will do the unpublished block but same thing also happens. Now theres no more elements left in the each, so it will exit out. It will try to create the flash messages, but since `errors.count > 0` and `success_ids.count > 0` are both false no flash messages are created.  So as pointed out above, therefore the user gets no flash message and they get shown nothing. Is this a desired behaviour?